src/api.js

```diff
@@ -88,16 +93,21 @@ builder.declare({
     res.writeHead(200, {
       'Content-Type': 'text/event-stream',
       'Cache-Control': 'no-cache',
+      'Access-Control-Allow-Origin': '*',
     });
     headWritten = true;
 
     let json_bindings = await parseAndValidateBindings(req.query.bindings);
     debug('Bindings parsed');
     var listener = await this.listeners.createListener(json_bindings);
-    sendEvent('ready');
-    const idleMessage = {code:404, message:'No messages received for 20s. Aborting...'};
-    let idleTimeout = setTimeout(() => abort(idleMessage), 20*1000);
-    
+
+    listener.resume().then(() => {
+      sendEvent('ready');
+      idleTimeout = setTimeout(() => abort(idleMessage), 20*1000);
```

Comment:
L114 seems to be creating a similar timeout. Could we have those places call `createIdleTimeout` instead to remove duplication? Also this will ensure we don't forget to clear the timeout before creating a new one.

Reply:
Yeah. Also I was thinking of increasing the timeout to 120seconds. 20s seems too short to me now.